### DISCUSSION TITLE
ZigBee 4 Button Remote - Usage in Rules

### DISCUSSION THREAD

**Discussion by malwikl** [arendst/Tasmota#13538] [Q&A] [answered]
Hi all,

I bought a 4 button remote from Tuya (Details can be found [here](https://de.aliexpress.com/item/1005002915179764.html?spm=a2g0o.detail.1000060.2.2ed3e0ec1787al&gps-id=pcDetailBottomMoreThisSeller&scm=1007.13339.169870.0&scm_id=1007.13339.169870.0&scm-url=1007.13339.169870.0&pvid=eac98623-45ee-49b9-8bc4-267af9267173&_t=gps-id:pcDetailBottomMoreThisSeller,scm-url:1007.13339.169870.0,pvid:eac98623-45ee-49b9-8bc4-267af9267173,tpp_buckets:668%232846%238108%231977&&pdp_ext_f=%7B%22sku_id%22:%2212000022765232798%22,%22sceneId%22:%223339%22%7D)).

Pairing works fine with tasmota2mqtt on a SonOff Zigbee Bridge, also the commands are received. What I am struggeling with, how to use the remote in rules. The received codes looks something like:
"ZBButton4":{"Device":"0x3B17","Name":"ZBButton4","0006!FD":"00","LidlPower":0,"Endpoint":2,"LinkQuality":126}
"ZBButton4":{"Device":"0x3B17","Name":"ZBButton4","0006!FD":"01","LidlPower":1,"Endpoint":2,"LinkQuality":126}

So Endpoint identifies the pressed button (1,2,3 or 4) and LidlPower if it is a single press or double press (0:Single, 1: Double).

Is there a way to use this in rules and have a different trigger per button and click type? Tried to figure it out myself, but did not find any hint...

Thanks,
MAlWiKl

**Comment by ascillato** [comment, marked as answer]
Hi,

You can trigger a rule using that nested JSON as:

```js
Rule1 1
Rule1
  ON ZbReceived#ZBButton4#LidlPower DO Var4 %value% ENDON
  ON ZbReceived#ZBButton4#Endpoint DO Event Button%value%Clicked=%var4% ENDON
  ON Event#Button1Clicked=0 DO [Any_command_for_button1_clicked_once] ENDON
  ON Event#Button1Clicked=1 DO [Any_command_for_button1_clicked_twice] ENDON
  ON Event#Button2Clicked=0 DO [Any_command_for_button2_clicked_once] ENDON
  ON Event#Button2Clicked=1 DO [Any_command_for_button2_clicked_twice] ENDON
  ...
```

Remember that for copying into the console everything must goes in one line like:
```js
Rule1 1
Rule1 ON ZbReceived#ZBButton4#LidlPower DO Var4 %value% ENDON ON ZbReceived#ZBButton4#Endpoint DO Event Button%value%Clicked=%var4% ENDON ON Event#Button1Clicked=0 DO [Any_command_for_button1_clicked_once] ENDON ...
```

**Reply by malwikl** [reply]
Works like a charm! Perfect, thanks a lot!!!
